### PR TITLE
Simplify temp table mgmt

### PR DESCRIPTION
**Pull request by akshatsikarwar**
Remove unnecessary complexity caused by sharing tmptable hash  between SQL threads.

### CODE REVIEW

File: db/sql.h [hunk 1]
```diff
@@ -752,6 +748,15 @@ struct query_path_component {
     LINKC_T(struct query_path_component) lnk;
 };
 
+struct temptable {
+    int rootpage;
+    struct temp_cursor *cursor;
+    struct temp_table *tbl;
```
Comment (nirbhayc) [inline db/sql.h]: IMO, `struct temptable` vs `struct temp_table` is a bit confusing.

Comment (akshatsikarwar) [reply db/sql.h]: Agreed. I can cleanup in subsequent PR (struct wasn't introduced in the PR, that is what it was called already).

Comment (akshatsikarwar) [reply db/sql.h]: I remember when both the structs were called `struct temp_table`. One day I couldn't take it anymore and renamed the one in `sqlglue.c`!

Comment (nirbhayc) [reply db/sql.h]: https://news.ycombinator.com/item?id=14598201

Comment (akshatsikarwar) [reply db/sql.h]: Indeed!

**Pull request merged**